datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxBackupRestoreService.java

```diff
@@ -560,31 +560,39 @@ private AttemptResult<DatalakeBackupStatusResponse> sdxFullDrSucceeded(SdxCluste
     public boolean shouldSdxBackupBePerformed(SdxCluster cluster, boolean entitlementEnabled) {
         String reason = null;
         if (!entitlementEnabled) {
-            reason = "Required entitlement for backup during this operation not enabled for this account.";
-        } else if (!datalakeDrConfig.isConfigured()) {
-            reason = "Datalake DR is not configured!";
-        } else {
-            DetailedEnvironmentResponse environmentResponse = environmentClientService.getByName(cluster.getEnvName());
-            CloudPlatform cloudPlatform = CloudPlatform.valueOf(environmentResponse.getCloudPlatform());
-            if (CloudPlatform.MOCK.equalsIgnoreCase(cloudPlatform.name())) {
-                return true;
-            }
+            LOGGER.info("Backup not triggered. Reason: Required entitlement for backup during this operation not enabled for this account");
```

Comment:
Is there a reason for this modification? Seems functionally equivalent but not necessarily more simple.

Reply:
why I believe this is simpler:
- removed several nested _if_ and put them on top level (inside original else block line 566).
- once error found we can see flow ends instead of eye-scanning till the end of the method
- eliminate variable _reason_ which is assigned only once (but this is not obvious)
- I personally don't like to make a decision of success/false based on string emptiness since a _reason_ is a supplementary information to the fact of an error. 

If you still find this is less understandable I will rollback - readability has high priority

Reply:
No this is fine thank you Sergiy for explaining yourself so well, I agree with your points now they just weren't clear on first glance.